### PR TITLE
Add `rz_strbuf_strip` to allow stripping end of `RzStrBuf`

### PR DESCRIPTION
**Pull request by DMaroo**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This utility function will be useful in general, but presently it is being used in #1678.

**Test plan**

Added unit tests. The unit tests are passing. CI should also pass, since no existing API has been changed.

**Closing issues**

None.

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: librz/include/rz_util/rz_strbuf.h [hunk 1]
```diff
@@ -26,6 +26,7 @@ RZ_API bool rz_strbuf_append_n(RzStrBuf *sb, const char *s, size_t l);
 RZ_API bool rz_strbuf_prepend(RzStrBuf *sb, const char *s);
 RZ_API bool rz_strbuf_appendf(RzStrBuf *sb, const char *fmt, ...) RZ_PRINTF_CHECK(2, 3);
 RZ_API bool rz_strbuf_vappendf(RzStrBuf *sb, const char *fmt, va_list ap);
+RZ_API bool rz_strbuf_strip(RzStrBuf *buf, int n);
```
Comment (wargio) [inline librz/include/rz_util/rz_strbuf.h:29]: i do not like the name, because i expect `rz_strbuf_strip` and `rz_strbuf_trim` to do the same thing (i.e. remove whitespaces from header and tail). i would suggest to rename it as `rz_strbuf_slice` and add start and end?

```c
RZ_API bool rz_strbuf_slice(RzStrBuf *buf, size_t start, size_t end);
```

Also @DMaroo avoid using `int` for sizes. use `size_t`

**Review by wargio — CHANGES_REQUESTED** [review]

File: librz/include/rz_util/rz_strbuf.h (hunk 1 above)
Comment (wargio) [reply librz/include/rz_util/rz_strbuf.h:29]: you can also add in the header
```c
#define rz_strbuf_slice_tail(buf, end) rz_strbuf_slice(buf, 0, end)
#define rz_strbuf_slice_head(buf, start) rz_strbuf_slice(buf, start, 0)
```

Comment (DMaroo) [reply librz/include/rz_util/rz_strbuf.h:29]: A `rz_strbuf_slice` function already exists.

https://github.com/rizinorg/rizin/blob/dev/librz/util/strbuf.c#L111-L130

It slices the `RzStrBuf` from `from` till `from + len`. The only reason why I created the strip function was that I needed to remove the trailing commas in the type representation of enums (#1678). I could also use this function, but for that I would have to manually access the length member of the `RzStrBuf`. To avoid that, I just created a tail strip function. What shall I do? 

TLDR: `rz_strbuf_slice` will also work in my case, but it will require me to access `RzStrBuf.len` explicitly.

Comment (ret2libc) [reply librz/include/rz_util/rz_strbuf.h:29]: you can use rz_strbuf_length function. I did not notice there is already `slice` API. If so, I think we don't need this new API. Is using `rz_strbuf_length` ok for you @DMaroo ?

Comment (DMaroo) [reply librz/include/rz_util/rz_strbuf.h:29]: Yup, makes sense. No need to have such a specific function. I'll close this PR then, and remove the `rz_strbuf_strip` commits from the other PR.
